#+title: GEBVeR

Genomic Estimated Breeding Values in R

A suite of tools for GEBV model cross evaluation, training, prediction and ground truth evaluation.

https://github.com/vangiangtran/BWGS

G. Charmet, L.G. Tran, J. Auzanneau and R. Rincent, S. Bouchet (2020) BWGS: A R package for genomic selection and its application to a wheat breeding programme. PLOS ONE 15(4): e0222733. https://doi.org/10.1371/journal.pone.0222733

V.G. Tran, D. Ly, G. Charmet. A Breed Wheat Genomic Selection pipeline. EUCARPIA Cereal Section & ITMI, 2014, Wernigerode, Germany. 359 p., 2014, Cereals for Food, Feed and Fuel – Challenge for Global Improvement. ⟨hal-02739684⟩

G. Charmet, V.G. Tran, D. Ly and J. Auzanneau. Breeding value estimation for yield and quality traits in wheat using BWGS pipeline. 9. Intern. Wheat conference, Sep 2015, Sydney, Australia. ⟨hal-02340476)

G. Charmet, V.G. Tran, D. Ly, J. Auzanneau. A R-based integrated pipeline for genomic and application to a wheat breeding programme. 20. Eucarpia General Congress, The European Plant Science Organisation (EPSO). BEL., Aug 2016, Zurich, Switzerland. ⟨hal-02743844⟩

J. Charmet, V.G. Tran, J. Auzanneau. Breeding value estimation for Yield and Quality traits using BWGS pipeline. JOBIM : Journées Ouvertes Biologie Informatique Mathématiques, Jul 2015, Clermont-Ferrand, France. ⟨hal-02299627⟩

S. Babi, R. Rincent, V.G. Tran, D. Alvarez, J. Bordes, et al.. Peut-on prédire la composition de la farine du blé?. 5. Colloque du Réseau Français de Biologie des Graines, Oct 2015, Clermont-Ferrand, France. ⟨hal-02299641⟩

* crossValidate

Purpose: Performs genomic prediction using various methods and
cross-validation, comparing their predictive abilities for a given phenotypic
trait using genotype data. The program provides a comprehensive analysis of
genomic prediction methods for the given dataset, allowing researchers to
compare and select the most suitable approach for their breeding program.

Requirements:
- R environment with packages: BWGS, vcfR, sys
- Input directory containing:
  - Uncompressed .vcf genotype files
  - CSV file with phenotype data (BLUE/BLUP values)

Usage:
1. Set working directory
2. Interactively select input files when prompted
3. Select column with genotypes
4. select the set of methods you will cross validate
5. Run the loop to calculate individual model results
6. Run the loop to collect and compare results

Function:
1. Sets up directory structure
2. Loads and processes genotype and phenotype data
3. Performs cross-validation using multiple user selected genomic prediction methods
4. Calculates predictive abilities and standard deviations for each method
5. Visualizes results through boxplots and bar plots

Output:
- CVcomparison.pdf: Boxplot comparing predictive abilities of methods
- CVdeviation.pdf: Bar plot of standard deviations for each method
- Summary.txt: Text file with summary statistics for all methods
- User may select a specific model for the following
- SummaryGBLUP.txt: Detailed summary for GBLUP method
- Summary_BA_BC_BL.txt: Detailed summaries for BA, BC, and BL methods
- Workspace image saved as "statePostCV" in data directory

* gtMetric

* phenoCorrelation

* predict

Purpose:
Requirements:
Usage:
Function:
Output:
